ghcide/src/Development/IDE/LSP/Outline.hs

```diff
@@ -46,13 +46,7 @@ moduleOutline ideState DocumentSymbolParams{ _textDocument = TextDocumentIdentif
                    (defDocumentSymbol l :: DocumentSymbol)
                      { _name  = pprText m
                      , _kind  = SkFile
-                     , _range = Range (Position 0 0) (Position 2147483647 0) -- _ltop is 0 0 0 0
-                     -- In the lsp spec from 3.16 Position takes a uinteger,
-                     -- where uinteger is 0 - 2^31 - 1. lsp-types currently has the type of line
-                     -- as Int. So instead of using `maxBound :: Int` we hardcode the maxBound of
-                     -- uinteger. 2 ^ 31 - 1 == 2147483647
```

Comment:
Uh oh. I think this means I got it wrong: `uinteger` according to the LSP specification is not, in fact a 32-bit unsigned integer, but a 31-bit unsigned integer. Presumably because it's actually a 32-bit signed integer that they just restrict to be positive. Terrible. I don't even know what to do about that.

Reply:
I made them an issue: https://github.com/microsoft/language-server-protocol/issues/1394

But in the short term this kind of sucks...

Reply:
I guess I can go back and use a custom type instead of `Word32` in `lsp`, sigh.

Reply:
Ouch, but we can go ahead with this lsp version? or you want to redefine the type in lso before the release
I guess we can continue using the hardcoded maxBound?

Reply:
It would be nice not to release a version of `lsp` that disruptively requires people to change their code and then doesn't even fix the problem :(

I think the best thing might be to change to a custom type in `lsp` indeed. It shouldn't affect this PR _too_ much. Either that or we should just revert the change in lsp.

Reply:
hmm but if we use UInt31 in LSP with all the required intances, including from integral, would no be matter of replacing Word32 with UInt31 here?

Reply:
Yes, that's basically it. I just need to do it. I'll try and find time tomorrow.

Reply:
maybe with a more generic name like LSPUInt? not sure if the data type could even change in the spec but I can imagine it could be bigger

Reply:
https://github.com/haskell/lsp/pull/382